src/Deprecated80/Integer.extension.st

```diff
@@ -0,0 +1,46 @@
+Extension { #name : #Integer }
+
+{ #category : #'*Deprecated80' }
+Integer >> digitAt: index [
+
+	self
+		deprecated: 'Please use byteAt: instead.'
+		transformWith:
+			'`@receiver digitAt: `@statements1'
+				-> '`@receiver byteAt: `@statements1'.
+	
+	^ self byteAt: index
+]
+
+{ #category : #'*Deprecated80' }
+Integer >> digitAt: index put: value [ 
+
+	self
+		deprecated: 'Please use #byteAt:put: instead'
+		transformWith:
+			'`@receiver digitAt: `@statements1 put: `@statements2'
+		   -> '`@receiver byteAt: `@statements1 put: `@statements2'.
+
+	^ self byteAt: index put: value 
+]
+
+{ #category : #'*Deprecated80' }
+Integer >> digitCompare: arg [ 
```

Comment:
I did not know about this method. It seems it is same as #threeWayCompareTo: which was introduced for sorting. So I think we just need to use #threeWayCompareTo and do not keep a new one. But please alter the implementation to use optimized version from #digitCompare:.

Reply:
And It will move threeWayCompare to kernel

Reply:
The main difference is that it calls a primitive by default ;-)

Reply:
Yes. I mean let's change threeWayCompare to use digitCompare code. It will work faster

Reply:
I opened an issue for that: https://github.com/pharo-project/pharo/issues/4605